test/built-ins/TypedArray/prototype/sort/comparefn-same-return-unsort-values.js

```diff
@@ -0,0 +1,34 @@
+// Copyright (C) 2016 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-%typedarray%.prototype.sort
+description: Order is the same if comparefn always returns the same value
+info: >
+  22.2.3.26 %TypedArray%.prototype.sort ( comparefn )
+
+  When the TypedArray SortCompare abstract operation is called with two
+  arguments x and y, the following steps are taken:
+
+  ...
+includes: [testTypedArray.js, compareArray.js]
+---*/
+
+var comparefn = function() {
+  return 0;
+};
+
+testWithTypedArrayConstructors(function(TA) {
+  var arr, sample;
+
+  arr = [4, 3, 2, 1];
+  sample = new TA(arr).sort(comparefn);
+  assert(compareArray(sample, arr), "[4, 3, 2, 1]");
```

Comment:
We may not be able to assert this. From Array.prototype.sort:

  > The sort is not necessarily stable (that is, elements that compare equal do
  > not necessarily remain in their original order).

  and further down:

  > Unless the sort order is specified above to be implementation-defined, the
  > returned object must have the following two characteristics:
  >
  > - There must be some mathematical permutation π of the nonnegative integers
  >   less than len, such that for every nonnegative integer j less than len,
  >   if property old[j] existed, then new[π(j)] is exactly the same value as
  >   old[j].  But if property old[j] did not exist, then new[π(j)] does not
  >   exist. 
  > - Then for all nonnegative integers j and k, each less than len, if
  >   SortCompare(old[j], old[k]) < 0 (see SortCompare below), then new[π(j)] <
  >   new[π(k)].